src/visualization.cpp

```diff
@@ -23,7 +23,7 @@ cv::Point2d worldToCanvas(const geo::Vector3& p)
 
 // ----------------------------------------------------------------------------------------------------
 
-void visualize(const World& world, const std::vector<RobotPtr>& robots, bool collision = false, bool show_full_map = false, Bbox centerframe = {-1e5, -1e5, 1e5, 1e5}, double robotRadius = 0.17)
+void visualize(const World& world, const Robot& robotobj, bool collision = false, bool show_full_map = false, Bbox centerframe = {-1e5, -1e5, 1e5, 1e5}, double robotRadius = 0.17)
```

Comment:
I would still go for `robot`. People touching this code should know that class names are PascalCase and instance/argument names are snake_case.

Make this consistent. Even in this function it is all mixed.
```suggestion
void visualize(const World& world, const Robot& robot, bool collision = false, bool show_full_map = false, Bbox center_frame = {-1e5, -1e5, 1e5, 1e5}, double robot_radius = 0.17)
```